llvm/tools/llvm-reduce/deltas/ReduceDistinctMetadata.cpp

```diff
@@ -0,0 +1,146 @@
+//===- ReduceDistinctMetadata.cpp - Specialized Delta Pass
+//------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------------------------------===//
+//
+// This file implements two functions used by the Generic Delta Debugging
+// Algorithm, which are used to reduce unnamed distinct metadata nodes.
+//
+//===------------------------------------------------------------------------------===//
+
+#include "ReduceDistinctMetadata.h"
+#include "Delta.h"
+#include "llvm/ADT/Sequence.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/IR/InstIterator.h"
+#include <algorithm>
+#include <queue>
+#include <set>
+
+using namespace llvm;
+
+// Traverse the graph breadth-first and try to remove unnamed metadata nodes
+void BFS_removal(MDNode *root,
+                 std::set<std::pair<unsigned int, MDNode *>> &nodesToDelete,
+                 MDNode *tmp, Oracle &O, Module &Program) {
+  std::queue<MDNode *> q{};
+  std::set<MDNode *>
+      visited{}; // Keep track of visited nodes not to get into loops
```

Comment:
Please follow the [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#id43), e.g. when it comes to variable names.